yjit/src/core.rs

```diff
@@ -179,7 +180,7 @@ pub struct Context
     sp_offset : i16,
 
     // Depth of this block in the sidechain (eg: inline-cache chain)
-    chain_depth: u8,
+    pub chain_depth: u8,
```

Comment:
Would rather we add getters as appropriate, especially given these values should remain constant.

Reply:
I think we can make a small effort to clean up some things as we go. There's already `get_sp_offset()` and `set_sp_offset()` methods for instance. As far as I can tell, we only ever set the chain depth to 0, so we could have a method called `reset_chain_depth()` that just sets it to zero.

Reply:
It seems nice to be able to put some logic in setters to limit how they can be used. You originally suggested we use rust so we can improve safety.